### DISCUSSION TITLE
external libraries vs. iOS app sync

### DISCUSSION THREAD

**Discussion by boku42** [immich-app/immich#14910] [Q&A]
Dear all
during my first test installation I learned that external libraries were introduced in the recent past. I also understood that external libraries are the recommended approach instead of immich-managed internal libraries. On the other hand, the iOS app does not support uploading photos into external libraries and since tools like synching do not seem to work for iOS photos, I decided not to use immich in the first place - although I would love to.
Now, I am in the phase of migrating photos once again and I would like to understand if new recommendations exist or new features are on their way. What is the best possible setup for immich and iOS  for now and the near future?
thank you, Patric

**Comment by mmomjian** [comment]
External libraries are not the recommended ideal approach. I’m not sure where you read that. The best feature set and support will always be for the built in library.  You should install the app and allow it to upload the pictures as designed.

**Reply by boku42** [reply]
oh, thank you, Matthew. Maybe I heard that in a YouTube video while searching for help on migration. I cannot use the app for upload since I already migrated the photos from Apple app to Synology Photos. Would you say, that the immesh upload program is the best option for migration server-side? It worked for my with 10K photos but not for my wife with 25K photos
